components/script/dom/clipboard.rs

```diff
@@ -95,6 +172,71 @@ impl ClipboardMethods<crate::DomTypeHolder> for Clipboard {
     }
 }
 
+impl RoutedPromiseListener<Result<String, String>> for Clipboard {
+    fn handle_response(
+        &self,
+        response: Result<String, String>,
+        promise: &Rc<Promise>,
+        _can_gc: CanGc,
+    ) {
+        let trusted_promise = TrustedPromise::new(promise.clone());
+        let text = response.unwrap_or_default();
+
+        // Step 3.4 Queue a global task on the clipboard task source,
+        // given realm’s global object, to perform the below steps:
+        self.global()
+            .task_manager()
+            .clipboard_task_source()
+            .queue(task!(read_text: move || {
+                let promise = trusted_promise.root();
+                let global = promise.global();
+
+                // Step 3.4.1 For each systemClipboardItem in data:
+                // Step 3.4.1.1 For each systemClipboardRepresentation in systemClipboardItem:
+                // TODO: Arboard provide the first item that has a String representation
+
+                // Step 3.4.1.1.1 Let mimeType be the result of running the
+                // well-known mime type from os specific format algorithm given systemClipboardRepresentation’s name.
+                // Note: This is done by arboard, so we just convert the format to a MIME
+                let mime_type = Mime::from_str("text/plain").unwrap();
+
+                // Step 3.4.1.1.2 If mimeType is null, continue this loop.
+
+                // Step 3.4.1.1.3 Let representation be a new representation.
+                let representation = Representation {
+                    mime_type,
+                    is_custom: false,
+                    data: Promise::new_resolved(&global, GlobalScope::get_cx(), DOMString::from(text), CanGc::note()),
```

Comment:
Note on the can gc here and elsewhere in `handle_response`: they should be passed to `handle_reponse`, see https://github.com/servo/servo/issues/36825